metadata-ingestion/src/datahub/ingestion/source/looker/looker_template_language.py

```diff
@@ -335,6 +340,54 @@ def _apply_transformation(self, value: str, view: dict) -> str:
         return self._apply_regx(value)
 
 
+class LookmlConstantTransformer(LookMLViewTransformer):
+    """
+    Replace LookML constants @{constant} from the manifest/configuration.
+    """
+
+    CONSTANT_PATTERN = r"@{(\w+)}"  # Matches @{constant}
+
+    def resolve_lookml_parameter(self, text: str) -> str:
+        """
+        Resolves LookML constants (@{ }) from manifest or config.
+        Logs warnings for misplaced or missing variables.
+        """
+
+        def replace_constants(match):
+            key = match.group(1)
+            if self.looker_constant:
+                value = next(
+                    (
+                        item["value"]
+                        for item in self.looker_constant
+                        if item["name"] == key
+                    ),
+                    None,
+                )
+                if value:
+                    return value
+
+            # Resolve constant
+            if key in self.source_config.lookml_constant:
+                return str(self.source_config.lookml_constant.get(key))
+
+            # Check if it's a misplaced liquid variable
+            if key in self.source_config.liquid_variable:
+                logger.warning(
+                    f"Misplaced liquid variable '@{{{key}}}' detected. Use '${{{key}}}' instead."
```

Comment:
this doesn't make sense - we should tell them to use `looker_constants` instead of `liquid_variables`